examples/experimental/dagster-airlift/dagster_airlift/in_airflow/dagster_operator.py

```diff
@@ -66,16 +77,67 @@ def compute_fn() -> None:
             )
             run_status = response.json()["data"]["runOrError"]["status"]
             if run_status in ["SUCCESS", "FAILURE", "CANCELED"]:
-                print(f"Run {run_id} completed with status {run_status}")  # noqa: T201
+                logger.debug(f"Run {run_id} completed with status {run_status}")
                 completed_runs[run_id] = run_status
     non_successful_runs = [
         run_id for run_id, status in completed_runs.items() if status != "SUCCESS"
     ]
     if non_successful_runs:
         raise Exception(f"Runs {non_successful_runs} did not complete successfully.")
-    print("All runs completed successfully.")  # noqa: T201
+    logger.debug("All runs completed successfully.")
     return None
 
 
-def build_dagster_task(task_id: str, dag: DAG, **kwargs):
-    return PythonOperator(task_id=task_id, dag=dag, python_callable=compute_fn, **kwargs)
+class DagsterOperator(PythonOperator):
+    pass
+
+
+def build_dagster_task(original_task: BaseOperator) -> DagsterOperator:
+    return instantiate_dagster_operator(original_task)
+
+
+def instantiate_dagster_operator(original_task: BaseOperator) -> DagsterOperator:
+    base_operator_args, base_operator_kwargs = get_args_and_kwargs(BaseOperator.__init__)
+    init_kwargs = {}
+
+    ignore_args = ["kwargs", "args", "dag"]
+    for arg in base_operator_args:
+        if arg in ignore_args or getattr(original_task, arg, None) is None:
+            continue
+        init_kwargs[arg] = getattr(original_task, arg)
+    for kwarg, default in base_operator_kwargs.items():
+        if kwarg in ignore_args or getattr(original_task, kwarg, None) is None:
+            continue
+        init_kwargs[kwarg] = getattr(original_task, kwarg, default)
+
+    return DagsterOperator(**init_kwargs, python_callable=compute_fn)
+
+
+def get_args_and_kwargs(func: Callable[..., Any]) -> Tuple[Set[str], Dict[str, Any]]:
```

Comment:
` get_args_and_kwargs`

This name is confusing because one assumes that is describing each element of tuple, but that is not what is happening here.

I think having a named_tuple called `BaseOperatorArgs` and then calling it `get_base_operator_args` or something.

This function also scares me a bit in terms of version compat etc. We are making a bunch of assumptions (namely that any transformation done in the base operator constructor is idempotent). I think it is worth a healthy docblock to explain all of that.

Reply:
That's right, I think the main assumption is exactly what you mention and it's not guaranteed to be true. While we do passthrough arguments that we can't find, I haven't figured out a good way to handle that case yet.

The more idiomatic thing to do might be to have some sort of airflow version -> accepted arguments matrix or something. But this is the "lazy" approach.

Reply:
That's right, I think the main assumption is exactly what you mention and it's not guaranteed to be true. While we do passthrough arguments that we can't find, I haven't figured out a good way to handle that case yet.

The more idiomatic thing to do might be to have some sort of airflow version -> accepted arguments matrix or something. But this is the "lazy" approach.

Reply:
That's right, I think the main assumption is exactly what you mention and it's not guaranteed to be the case that arguments->properties are idiomatic. While we do passthrough arguments that we can't find, I haven't figured out a good way to handle that case yet.

The more idiomatic thing to do might be to have some sort of airflow version -> accepted arguments matrix or something. But this is the "lazy" approach.

Reply:
I think my preference would be to get this approach checked in so that it's in the git history, and then we can simplify if we feel it's necessary.

Reply:
I think we should a better name at a minimum as I think the current name is actively misleading.

Reply:
Regarding the function name though, I'm a bit confused what you mean. I think this is basically describing the return type of each element of the tuple. get_args_and_kwargs is agnostic to the function passed in, so I would feel weird calling it `get_base_operator_args`. Perhaps some of the confusion is coming from the names of variables here `args_with_defaults` etc, since `args_with_defaults` really means `kwargs`. I'll rename variables and see if that makes it more clear what's going on.

Reply:
per offline discussion rename to `get_params`